pkg/operation/botanist/component/extensions/bastion/bastion.go

```diff
@@ -0,0 +1,190 @@
+) Interface {
+	return &bastion{
+		client:              client,
+		logger:              logger,
+		values:              values,
+		waitInterval:        waitInterval,
+		waitSevereThreshold: waitSevereThreshold,
+		waitTimeout:         waitTimeout,
+	}
+}
+
+type bastion struct {
+	values              *Values
+	logger              logrus.FieldLogger
+	client              client.Client
+	waitInterval        time.Duration
+	waitSevereThreshold time.Duration
+	waitTimeout         time.Duration
+}
+
+// Deploy uses the seed client to create or update the Bastion custom resource in the Seed.
+func (b *bastion) Deploy(ctx context.Context) error {
+	_, err := b.deploy(ctx)
+	return err
+}
+
+func (b *bastion) deploy(ctx context.Context) (extensionsv1alpha1.Object, error) {
+	bastion := &extensionsv1alpha1.Bastion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: b.values.Name,
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, b.client, bastion, func() error {
+		extensionsIngress := make([]extensionsv1alpha1.BastionIngressPolicy, len(b.values.Ingress))
+		for i, ingress := range b.values.Ingress {
+			extensionsIngress[i] = extensionsv1alpha1.BastionIngressPolicy{
+				IPBlock: ingress.IPBlock,
+			}
+		}
+
+		bastion.Spec = extensionsv1alpha1.BastionSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type:           b.values.Type,
+				ProviderConfig: b.values.ProviderConfig,
+			},
+			UserData: b.values.UserData,
+			Ingress:  extensionsIngress,
+		}
+
+		return nil
+	})
+
+	return bastion, err
+}
+
+// Destroy deletes the Bastion CRD
+func (b *bastion) Destroy(ctx context.Context) error {
+	return extensions.DeleteExtensionCR(
+		ctx,
+		b.client,
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Bastion{} },
+		"",
+		b.values.Name,
+	)
+}
+
+// Wait waits until the Bastion CRD is ready (deployed or restored)
+func (b *bastion) Wait(ctx context.Context) error {
+	return extensions.WaitUntilExtensionCRReady(
+		ctx,
+		b.client,
+		b.logger,
+		func() client.Object { return &extensionsv1alpha1.Bastion{} },
+		extensionsv1alpha1.BastionResource,
+		"",
+		b.values.Name,
+		b.waitInterval,
+		b.waitSevereThreshold,
+		b.waitTimeout,
+		nil,
+	)
+}
+
+// WaitCleanup waits until the Bastion CRD is deleted
+func (b *bastion) WaitCleanup(ctx context.Context) error {
+	return extensions.WaitUntilExtensionCRDeleted(
+		ctx,
+		b.client,
+		b.logger,
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Bastion{} },
+		extensionsv1alpha1.BastionResource,
+		"",
+		b.values.Name,
+		b.waitInterval,
+		b.waitTimeout,
+	)
+}
+
+func (b *bastion) SetType(t string) {
+	b.values.Type = t
+}
+
+func (b *bastion) SetProviderConfig(providerConfig *runtime.RawExtension) {
+	b.values.ProviderConfig = providerConfig
+}
+
+func (b *bastion) SetUserData(userData []byte) {
+	b.values.UserData = userData
+}
+
+func (b *bastion) SetIngress(ingress []gardencorev1alpha1.BastionIngressPolicy) {
+	b.values.Ingress = ingress
+}
```

Comment:
Why do you need those functions? I guess all of these values are already known by the time the `Interface` is instantiated?